### PR TITLE
consumer to pass in plugin presets themselves

### PR DESCRIPTION
**Pull request by gyfchong**
Kinda makes it clearer to consumers?

### CODE REVIEW

File: packages/package-bundler/src/presets/default/plugins-default.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      {
+        find: "__@cultureamp/i18n-react-intl/locales",
+        replacement: "locales",
+      },
+    ],
+  }),
+  // These libraries aren't used in KAIO, and require polyfills to be set up
+  // in consuming repos. Ignoring them here removes the need for extra setup in
+  // consuming repos.
+  ignore(["stream", "http", "https", "zlib"]),
```
Comment (jakubriedl) [inline packages/package-bundler/src/presets/default/plugins-default.ts:32]: should we include also other Nodejs modules similarly as we do for [example here](https://github.com/cultureamp/unified-home/blob/master/packages/next-storybook/src/addons/sensibleDefaults/preset.ts#L53-L66)?

File: packages/package-bundler/src/rollup.ts [hunk 2]
```diff
@@ -1,70 +1,25 @@
...
-      ignore(["stream", "http", "https", "zlib"]),
-      babel({ babelHelpers: "bundled" }),
-      getBabelOutputPlugin({
-        plugins: [
-          "@babel/plugin-transform-react-pure-annotations",
-          "babel-plugin-pure-static-props",
-        ],
-      }),
-    ],
+    plugins: [...config.plugins, ...presetDefault({ alias: config.alias })],
```
Comment (jakubriedl) [inline packages/package-bundler/src/rollup.ts:22]: any reason not to pass whole config to preset?
```suggestion
    plugins: [...config.plugins, ...presetDefault(config)],
```

Comment (gyfchong) [reply packages/package-bundler/src/rollup.ts:22]: Not technically, but I don't want to pass anything it doesn't need. Admittedly the lines are blurring here, cause if we send in all config, then we may as well pull all the config into this file again?

Comment (jakubriedl) [reply packages/package-bundler/src/rollup.ts:22]: My take is that it's matter of separating concerns/coupling
- you want the default preset to be in seperate file so it's more self sufficient
- also you don't want to assume from here what that file needs from the config

Comment (jakubriedl) [reply packages/package-bundler/src/rollup.ts:22]: also when I think about it, the default preset should be inside the ui library preset. So it allows consumer to eject from default if they need to

Comment (gyfchong) [reply packages/package-bundler/src/rollup.ts:22]: Hm yeah makes sense, so there's 3 options:
- UI Library preset (inherits default preset)
- Default Preset
- None

Comment (jakubriedl) [reply packages/package-bundler/src/rollup.ts:22]: yes

**Pull request merged**